mesonbuild/compilers/d.py

```diff
@@ -581,7 +585,7 @@ def _get_compiler_check_args(self, env, extra_args, dependencies, mode='compile'
         elif not isinstance(dependencies, list):
             dependencies = [dependencies]
         # Collect compiler arguments
-        args = self.compiler_args()
+        args = DCompilerArgs(self)
```

Comment:
Shouldn't `self.compiler_args()` still work, since you have overwritten it below?

Reply:
In current version, it does not.

From what I understood, compiler_args is overriden for "clike" languages, from which C/C++ inherits.
A mechanism was added in commit introducing regression where some options are defined as "deduplicated" or "prepended". I did the same kind of thing with D, by introducing DCompilerArgs (which inherits CompilerArgs).

Without this, unit test does not pass.
Maybe @dcbaker  who introduced all this has a comment (or better fix) for this.

Reply:
You did override `compiler_args` in `DCompiler`, so `self.compiler_args()` will use your version from `DCompiler`.

Reply:
Yes, just realised now I missed what you said.

You are totally right, I pushed a new version without this.